File: Python/Program/Class/Ram.py
from Class.PartKomputer import PartKomputer

class Ram(PartKomputer):
    def __init__(self, id=0, nama="", harga=0, stok=0,
                 no_part="", manufaktur="", kondisi="",
                 ddr_ver=0, memory=0, speed=0):
        super().__init__(id, nama, harga, stok, no_part, manufaktur, kondisi)
        self._ddr_ver = ddr_ver
        self._memory = memory
        self._speed = speed

    def set_ddr_ver(self, ddr_ver):
        self._ddr_ver = ddr_ver

    def get_ddr_ver(self):
        return self._ddr_ver

    def set_memory(self, memory):
        self._memory = memory

    def get_memory(self):
        return self._memory

    def set_speed(self, speed):
        self._speed = speed

    def get_speed(self):
        return self._speed

File: Java/Program/Class/Ram.java
package Class;

public class Ram extends PartKomputer {
    private int DDR_Ver;
    private int memory;
    private int speed;

    public Ram() {
        super();
        this.DDR_Ver = 0;
        this.memory = 0;
        this.speed = 0;
    }

    public Ram(int id, String nama, int harga, int stok,
               String noPart, String manufaktur, String kondisi,
               int DDR_Ver, int memory, int speed) {
        super(id, nama, harga, stok, noPart, manufaktur, kondisi);
        this.DDR_Ver = DDR_Ver;
        this.memory = memory;
        this.speed = speed;
    }

    public int getDDRVer() { return DDR_Ver; }
    public void setDDRVer(int DDR_Ver) { this.DDR_Ver = DDR_Ver; }

    public int getMemory() { return memory; }
    public void setMemory(int memory) { this.memory = memory; }

    public int getSpeed() { return speed; }
    public void setSpeed(int speed) { this.speed = speed; }
}
